src/offline/mod_offline.erl

```diff
@@ -313,14 +314,17 @@ code_change(_OldVsn, State, _Extra) ->
 %% This function should be called only from a hook
 %% Calling it directly is dangerous and may store unwanted messages
 %% in the offline storage (e.g. messages of type error)
--spec inspect_packet(mongoose_acc:t(), jid:jid(), jid:jid(), exml:element()) -> mongoose_acc:t().
-inspect_packet(Acc, From, To, Packet) ->
+-spec inspect_packet(Acc, Params, Extra) -> {ok | stop, Acc} when
+    Acc :: mongoose_acc:t(),
+    Params :: map(),
+    Extra :: map().
+inspect_packet(Acc, #{from := From, to := To, packet := Packet}, _Params) ->
```

Comment:
I see that you followed convention to name the last argument `_Extra`. Please rename this as well so it won't confuse anybody.